src/main/java/teammates/ui/webapi/action/DeleteCourseAction.java

```diff
@@ -21,23 +20,20 @@ public void checkSpecificAccessControl() {
         if (!userInfo.isInstructor) {
             throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
         }
-
         String idOfCourseToDelete = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
         gateKeeper.verifyAccessible(logic.getInstructorForGoogleId(idOfCourseToDelete, userInfo.id),
-                logic.getCourse(idOfCourseToDelete), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE);
+                logic.getCourse(idOfCourseToDelete),
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE);
     }
 
     @Override
     public ActionResult execute() {
         String idOfCourseToDelete = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-        try {
-            logic.moveCourseToRecycleBin(idOfCourseToDelete);
 
-            return new JsonResult("The course " + idOfCourseToDelete
-                    + " has been deleted. You can restore it from the Recycle Bin manually.");
-        } catch (EntityDoesNotExistException e) {
-            return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
-        }
-    }
+        CourseAttributes courseAttributes = logic.getCourse(idOfCourseToDelete);
+
+        logic.deleteCourse(idOfCourseToDelete);
 
+        return new JsonResult(new CourseData(courseAttributes));
```

Comment:
DELETE method should just a message right?

Reply:
A MessageOutput? I thought all API calls should be changed to not concern itself with what to be shown to the user.